src/lib/swapper/swappers/ThorchainSwapper/ThorchainSwapper.ts

```diff
@@ -1,59 +1,40 @@
+import type { StdSignDoc } from '@keplr-wallet/types'
 import type { AssetId } from '@shapeshiftoss/caip'
-import { CHAIN_NAMESPACE, fromAssetId, fromChainId, thorchainAssetId } from '@shapeshiftoss/caip'
-import type {
-  CosmosSdkChainAdapter,
-  SignTx,
-  UtxoChainAdapter,
-  UtxoChainId,
-} from '@shapeshiftoss/chain-adapters'
-import type { ThorchainSignTx } from '@shapeshiftoss/hdwallet-core'
+import { fromAssetId, thorchainAssetId } from '@shapeshiftoss/caip'
+import type { BTCSignTx } from '@shapeshiftoss/hdwallet-core'
 import { getConfig } from 'config'
-import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
 import {
   buySupportedChainIds,
   sellSupportedChainIds,
 } from 'lib/swapper/swappers/ThorchainSwapper/constants'
 import type { ThorChainId, ThornodePoolResponse } from 'lib/swapper/swappers/ThorchainSwapper/types'
 import { poolAssetIdToAssetId } from 'lib/swapper/swappers/ThorchainSwapper/utils/poolAssetHelpers/poolAssetHelpers'
 import { thorService } from 'lib/swapper/swappers/ThorchainSwapper/utils/thorService'
-import type { BuyAssetBySellIdInput, ExecuteTradeArgs, Swapper } from 'lib/swapper/types'
-import { assertUnreachable } from 'lib/utils'
-import { executeEvmTrade } from 'lib/utils/evm'
+import type {
+  BuyAssetBySellIdInput,
+  CosmosSdkTransactionExecutionProps,
+  Swapper,
+  UtxoTransactionExecutionProps,
+} from 'lib/swapper/types'
+import { executeEvmTrade2 } from 'lib/utils/evm'
 
 const daemonUrl = getConfig().REACT_APP_THORCHAIN_NODE_URL
 
 export const thorchainSwapper: Swapper = {
-  executeTrade: async ({ txToSign, wallet, chainId }: ExecuteTradeArgs): Promise<string> => {
-    const { chainNamespace } = fromChainId(chainId)
-    const chainAdapterManager = getChainAdapterManager()
-    const adapter = chainAdapterManager.get(chainId)
-
-    switch (chainNamespace) {
-      case CHAIN_NAMESPACE.Evm: {
-        return executeEvmTrade({ txToSign, wallet, chainId })
-      }
+  executeEvmTransaction: executeEvmTrade2,
 
-      case CHAIN_NAMESPACE.CosmosSdk: {
-        const cosmosSdkChainAdapter = adapter as unknown as CosmosSdkChainAdapter
-        const signedTx = await cosmosSdkChainAdapter.signTransaction({
-          txToSign: txToSign as ThorchainSignTx,
-          wallet,
-        })
-        return cosmosSdkChainAdapter.broadcastTransaction(signedTx)
-      }
-
-      case CHAIN_NAMESPACE.Utxo: {
-        const utxoChainAdapter = adapter as unknown as UtxoChainAdapter
-        const signedTx = await utxoChainAdapter.signTransaction({
-          txToSign: txToSign as SignTx<UtxoChainId>,
-          wallet,
-        })
-        return utxoChainAdapter.broadcastTransaction(signedTx)
-      }
+  executeCosmosSdkTransaction: async (
+    txToSign: StdSignDoc,
```

Comment:
absolute 🧠 , indeed, that's the payload which is going to come from Keplr - this may break for native / KK however?

Reply:
I'll get ops to test KK, native seems to be working